### PR TITLE
[Coral-756] Distribution of Spark Source

### LINKED ISSUE #756: Handling Spark Source in a distributed manner
The Spark word count example currently runs without distributing the source tasks, but performing it on the driver.
I think this issue could be handled in a couple of ways:

1. Getting the iterator as it is being done currently, and handling them in the executor.
2. Setting up SparkEnv in the executors and running the given RDDs there.
3. Setting up a whole new SparkSession in the executors in local mode, and running separate RDDs there.

This task was making the PR too large, and I think it would be more neat to handle it as a separate issue.
Follows #711, #745

### PR DESCRIPTION
**Pull request by wonook**
This PR:

- fixes the current driver-only implementation of Spark Source execution and makes it to be run on each executor instead.

resolves #756

### CODE REVIEW

File: examples/resources/sample_input_wordcount [hunk 1]
```diff
@@ -2,10 +2,10 @@ banana
 ski snowboard
 banana ski one two three
 bicycle piano piano three
-two three ski ski
-jack wonook wonook
-john john jooy jooy jooy wonook
-ski snowboard
-shakespeare sanha girl
-piano bicycle
+two three ski ski jangho
```
Comment (sanha) [inline examples/resources/sample_input_wordcount]: Can't we check the output validity?

File: compiler/frontend/spark/src/main/java/edu/snu/coral/compiler/frontend/spark/source/SparkBoundedSourceVertex.java [hunk 2]
```diff
@@ -60,25 +62,34 @@ public SparkBoundedSourceVertex getClone() {
...
-    public Iterable<T> read() {
-      new SparkContext(conf);
-      return iterable;
+    public Iterable<T> read() throws Exception {
+      final SparkSession spark = SparkSession.builder()
+          .config(sessionInitialConf)
+          .getOrCreate();
+      final Dataset<T> dataset = SparkSession.initializeDataset(spark, commands);
+
+      final RDD<T> rdd = dataset.rdd();
```
Comment (sanha) [inline compiler/frontend/spark/src/main/java/edu/snu/coral/compiler/frontend/spark/source/SparkBoundedSourceVertex.java]: Could you add some comments that this operation does not load the full dataset but only the handler of partitions?

File: examples/resources/sample_input_wordcount (hunk 1 above)
Comment (wonook) [reply examples/resources/sample_input_wordcount]: Sure. I'll add the logic to the test

File: compiler/frontend/spark/src/main/java/edu/snu/coral/compiler/frontend/spark/source/SparkBoundedSourceVertex.java (hunk 2 above)
Comment (wonook) [reply compiler/frontend/spark/src/main/java/edu/snu/coral/compiler/frontend/spark/source/SparkBoundedSourceVertex.java]: Sure. I'll do that.

**Pull request merged**